src/code/stores/simulation-store.coffee

```diff
@@ -66,12 +68,15 @@ SimulationStore   = Reflux.createStore
 
   onCollapseSimulationPanel: ->
     @settings.simulationPanelExpanded = false
+    SimulationActions.simulationEnded()
     @notifyChange()
 
   onGraphChanged: (data)->
     @nodes = data.nodes
     @settings.modelIsRunnable = @_checkModelIsRunnable()
     @settings.graphHasCollector = @_checkForCollectors()
+    if @settings.modelIsRunning
+      @_runSimulation()
```

Comment:
While it seems to work really well for the most part for `@_runSimulation` to be called on GraphChanged, there seem like there are a couple of edge-cases where this shouldn't happen, namely: adding and removing links without defined relationships (since this doesn't actually change the underlying graph). I don't know whether it would be possible for making a new link without a relationship to not call `onGraphChanged`, or if some other filter is needed.

The other place where this is trouble is undo/redo. If you move a slider, the simulation is run, and then if you undo that movement, this still triggers new data to be added, so the graph keeps moving forward. It seems like this would need to be special-cased some way, to correctly allow for undoing data additions, and maybe that's too much for this PR's scope.